src/views/List.jsx

```diff
@@ -1,23 +1,39 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+
+	const filterList = data.filter((item) => {
+		if (searchTerm === '') {
+			return item;
+		} else if (item.name.toLowerCase().includes(searchTerm.toLowerCase())) {
+			return item;
+		}
+	});
+
+	const renderList = filterList.map((item) => (
+		<ListItem name={item.name} key={item.id} />
+	));
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		setSearchTerm('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
-			<ul>
-				{
-					/**
-					 * TODO: write some JavaScript that renders the `data` array
-					 * using the `ListItem` component that's imported at the top
-					 * of this file.
-					 */
-					data.map((item) => (
-						<ListItem name={item.name} key={item.id} />
-					))
-				}
-			</ul>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="search-filter">Item name:</label>
+				<input
+					type="text"
+					id="search-filter"
+					value={searchTerm}
+					onChange={(e) => setSearchTerm(e.target.value)}
+				/>
+				<button type="submit"> X </button>
+			</form>
```

Comment:
This works well, you two. I was wondering what your thoughts are on future code organization. Do you think a reusable input component with a filter feature could be a thing? Why or why not? Definitely approved. Happy Friday!

Reply:
Hi Dana -- Thank you for bringing this up! We could definitely benefit from creating a reusable input component if we wanted to user the filter method on other pages.